### DISCUSSION TITLE
Why PhoneAuthOptions.newBuilder() has a setTimeout(30L, ...)?

### DISCUSSION THREAD

**Discussion by efstathiosntonas** [invertase/react-native-firebase#6971] [General]
Hey @mikehardy, hope you're doing well. Can you please explain why the [default setTimeout is set to 30L](https://github.com/invertase/react-native-firebase/blob/e118597e22d13355cd83fd3817dacfb6546eefa2/packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java#L1021)? The sdk docs have a [value of 60L](https://firebase.google.com/docs/auth/android/phone-auth#send-a-verification-code-to-the-users-phone) as an example.

Recently we are experiencing many sms expired errors and I was wondering if this value can affect this error. We have a countdown blocker of 30 seconds awaiting for auto-retrieval, when timeout ends and the auto-retrieval failed then the user can enter the code manually (I know this is not recommended but many users were spamming the "resend code" logic).

It could be beneficial to pass a prop for it so users can freely change this? 🤔 

Thanks!

**Comment by mikehardy** [comment]
Huh - seems like at a minimum, 60 seconds might be better, the 30 came in the original implementation from @fzuellich - perhaps they know?

I would be open to a PR that switches it to 60 (having it conform with the docs seems like a reasonable thing...) or with passing props and a default to 60 if no props passed

**Reply by efstathiosntonas** [reply]
Thanks for the reply Mike, I’ve ended up patching this value to 0, the amount of sms expired codes was really high. Was really hard to debug since some devices had no issue while others were constantly failing. 

At some point I was thinking it was a carrier issue since on my country the sms arrived at about 15-20secs while on other countries eg US the code arrived in device in <5 secs. 

I have checked my authStateChanged functions and logic 100 times, after setting it to 0 (like multi is doing) I’ve stopped receiving sms expired.

**Reply by efstathiosntonas** [reply]
Here's [multiFactor](https://github.com/firebase/quickstart-android/blob/047f84f68ac3f9a184a8785e5b66f54e7c5cf206/auth/app/src/main/java/com/google/firebase/quickstart/auth/java/MultiFactorEnrollFragment.java#L116) being set to `0L` on quickstart guide for Android.

**Reply by fzuellich** [reply]
I don't recall any specific reason for going with 30s 👍
